python/pyarrow/_parquet.pyx

```diff
@@ -505,6 +506,204 @@ cdef class ColumnChunkMetaData(_Weakrefable):
         return self.metadata.GetColumnIndexLocation().has_value()
 
 
+cdef class SortingColumn:
+    """
+    Sorting specification for a single column.
+
+    Returned by :meth:`RowGroupMetaData.sorting_columns` and used in
+    :class:`ParquetWriter` to specify the sort order of the data.
+
+    Parameters
+    ----------
+    column_index : int
+        Index of column that data is sorted by.
+    descending : bool, default False
+        Whether column is sorted in descending order.
+    nulls_first : bool, default False
+        Whether null values appear before valid values.
+
+    Notes
+    -----
+
+    Column indices are zero-based, refer only to leaf fields, and are in
```

Comment:
By the way, in current implemention, parquet itself will not check the `sorting_order` is satisfied in user input. So we'd better:

1. Maybe only enable writing for test?
2. Or tell the user this is so dangerous and user should set this order if he/she cannot not gurantee ordering.

Reply:
This is made clear in the docstrings of the Parquet Writer: https://github.com/apache/arrow/blob/5db4e8e243ec8ca6724252bca7f478a0c417e76e/python/pyarrow/parquet/core.py#L885-L888